Feature/921l/othersview

921l opened the pull request.

921l committed:
feat(pages): 新增他人视角页面

921l committed:
feat(pages): 新增他人视角页面

921l committed:
feat(pages): 新增他人视角页面

921l committed:
feat(pages): 新增他人视角页面

921l committed:
feat(pages): 新增他人视角页面

921l committed:
Merge branch 'dev' into feature/921l/Othersview

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
feat(pages): 新增个人页面及他人页面

Marstonchan commented on src/pages/profile/components/FansListItem.vue:
```diff
@@ -0,0 +1,109 @@
+    console.log(error);
+  }
+};
+getData();
```
这里放到生命周期中去执行。

Marstonchan commented on src/pages/profile/components/FansListItem.vue:
```diff
@@ -0,0 +1,109 @@
+getData();
+const switchState = () => {
+  if (userData.data.followed) {
+    othersViewService.unFollow(props.item.id).then(() => {
```
回合一下dev，首页也有这几个接口，用同一个就可以了，统一抽到 post_service 里面。

Marstonchan commented on src/pages/profile/components/FansListItem.vue:
```diff
@@ -0,0 +1,109 @@
+      userData.data.followed = false;
+    });
+  } else {
+    othersViewService.follow(props.item.id).then(() => {
```
统一使用 try/catch + async/await 来写接口请求吧，可读性更高一些

Marstonchan commented on src/pages/profile/components/PostTime.vue:
```diff
@@ -0,0 +1,50 @@
+const now = dayjs();
+const standardTimeDay = time.format('YYYY-MM-DD');
+const standardTimeYear = time.format('YYYY');
+dayjs.updateLocale('zh-cn', {
```
感觉这个里面是不是又会有之前在time-ago里遇到的问题？这些 time 之类的不是响应式的

Marstonchan commented on src/pages/profile/components/card.vue:
```diff
@@ -0,0 +1,147 @@
+import OthersViewService from '~/service/othersView_service';
+
+const othersViewService = new OthersViewService();
+const props = defineProps({
```
nice！都统一放到 props 和 state 里管理状态

Marstonchan commented on src/pages/profile/components/card.vue:
```diff
@@ -0,0 +1,147 @@
+const switchFabulousColor = () => {
+  if (!state.FabulousColor) {
+    try {
+      othersViewService.like(props.item.postId).then(
```
try-catch要和promise.then这样用，try-catch的话应该用 async/await

Marstonchan commented on src/pages/profile/components/card.vue:
```diff
@@ -0,0 +1,147 @@
+    }
+  } else {
+    try {
+      othersViewService.unlike(props.item.postId).then(() => {
```
这里也是，都改用和首页一样的接口吧

Marstonchan commented on src/pages/profile/components/card.vue:
```diff
@@ -0,0 +1,147 @@
+const switchStarColor = () => {
+  if (!state.StarColor) {
+    try {
+      othersViewService.favorite(props.item.postId).then(
```
同上

Marstonchan commented on src/pages/profile/components/constants.ts:
```diff
@@ -0,0 +1,8 @@
+export const LEVEL_COLOR_LIST = [
```
这个在 constants/user_info 里已经定义了，不需要重复定义，直接使用即可

Marstonchan commented on src/pages/profile/components/timeLine.vue:
```diff
@@ -0,0 +1,272 @@
+  },
+});
+const postData = {
+  data: {
```
这些 data 不需要套一层。直接从返回值里解构就可以了

Marstonchan replied:
这里的 postData 不是响应式的，真的没有问题吗？

Marstonchan commented on src/pages/profile/components/timeLine.vue:
```diff
@@ -0,0 +1,272 @@
+});
+const getData = () => {
+  try {
+    othersViewService.getPost(props.item.postId).then((res) => {
```
同上，改成async await

Marstonchan commented on src/pages/profile/components/timeLine.vue:
```diff
@@ -0,0 +1,272 @@
+const switchStarColor = () => {
+  if (!state.StarColor) {
+    try {
+      othersViewService.favorite(props.item.postId).then(
```
同上，使用同一个接口即可

Marstonchan commented on src/pages/profile/components/user.vue:
```diff
@@ -0,0 +1,143 @@
+import { CSSProperties, computed } from 'vue';
+import { LEVEL_COLOR_LIST } from './constants';
+
+export interface AvatarProps {
```
有类型定义很nice，但是尽量不要在组件里export类型，如果外面还有别的地方用的话，放到types/目录下

Marstonchan commented on src/pages/profile/components/user.vue:
```diff
@@ -0,0 +1,143 @@
+  bgColor?: string;
+}
+export interface UserInfoProps {
+  nickname: string;
```
可以扩展自 UserInfoFacade 吗？挺多字段是重复的

Marstonchan commented on src/pages/profile/components/user.vue:
vue 文件大驼峰命名

Marstonchan commented on src/pages/profile/components/card.vue:
vue文件大驼峰命名

Marstonchan commented on src/pages/profile/components/timeLine.vue:
vue文件大驼峰命名

Marstonchan commented on src/pages/profile/icon/female.svg:
这三个文件有使用到吗？用到的话放到 assets/icons 目录下，新建一下

Marstonchan commented on src/service/othersView_service.ts:
```diff
@@ -0,0 +1,71 @@
+
+export default class OthersViewService {
+  async getUserInfo(userId: Number) {
+    const data = await axios.get(`https://inxupt.alkaidchen.com:443/user/${userId}/page`);
```
三个问题：
1. 补充返回值类型定义，参考其他 service
2. 请求不需要写死绝对路径，`/user/${userId}/page` 即可，前面的域名和端口会在配置文件中定义，这么写死的话就没有办法动态区分环境了
3. service 层只返回从 `axios` 中返回的 `data` 即可，这样外面就不需要多定义一层 `data` 了，具体参考其他 service 实现

Marstonchan commented on src/service/othersView_service.ts:
```diff
@@ -0,0 +1,71 @@
+
+  async getFansInfo(userId: Number, pageNum: Number) {
+    const data = await axios.get(
+      `https://inxupt.alkaidchen.com:443/user/${userId}/fans?pageNum=${pageNum}`
```
同上，缺失类型定义

Marstonchan commented on src/service/othersView_service.ts:
```diff
@@ -0,0 +1,71 @@
+  }
+
+  async follow(userId: Number) {
+    const data = await axios.post(`https://inxupt.alkaidchen.com:443/user/follow/${userId}`);
```
和首页的service统一一下

Marstonchan commented on src/pages/profile/personView.vue:
```diff
@@ -0,0 +1,537 @@
+  scroll_top: 0,
+  goTop_show: false,
+  showPreview: false,
+  imgData: ['https://shijuepi.com/uploads/allimg/201222/1-2012221T114.jpg'],
```
这里写死的是什么图片？

Marstonchan commented on src/pages/profile/personView.vue:
```diff
@@ -0,0 +1,537 @@
+  imgData: ['https://shijuepi.com/uploads/allimg/201222/1-2012221T114.jpg'],
+});
+const userData = reactive({
+  data: {
```
不需要 `data: {}`

Marstonchan commented on src/pages/profile/personView.vue:
```diff
@@ -0,0 +1,537 @@
+  followsIsLoading: '',
+  postIsLoading: '',
+  favoriteIsLoading: '',
+  scroll_top: 0,
```
为啥还有下划线的命名。。。统一使用小驼峰命名

Marstonchan commented on src/pages/profile/personView.vue:
```diff
@@ -0,0 +1,537 @@
+const othersViewService = new OthersViewService();
+const getData = () => {
+  try {
+    othersViewService.getUserInfo(props.id).then((res) => {
```
同

Marstonchan commented on src/pages/profile/personView.vue:
```diff
@@ -0,0 +1,537 @@
+  }
+
+  try {
+    othersViewService.getFansInfo(props.id, state.fansPage).then((res) => {
```
同

Marstonchan commented on src/pages/profile/personView.vue:
```diff
@@ -0,0 +1,537 @@
+  }
+
+  try {
+    othersViewService.getfollowsInfo(props.id, state.followsPage).then((res) => {
```
把这些请求逻辑抽都分别抽离为不同的方法吧，不要在一个里面写这么多请求和 trycatch，只需要 await 即可

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
Rename card.vue to Card.vue

921l committed:
Rename timeLine.vue to TimeLine.vue

921l committed:
Rename user.vue to User.vue

921l committed:
Rename othersView.vue to OthersView.vue

921l committed:
Rename personView.vue to PersonView.vue

Marstonchan commented on src/types/person_types.ts:
```diff
@@ -0,0 +1,128 @@
+  total: number;
+}
+
+export interface FollowsInfoFacade {
```
用 `PagedResponseData` 这个泛型即可

Marstonchan commented on src/service/othersView_service.ts:
```diff
@@ -1,71 +1,48 @@
-    const data = await axios.get(
-      `https://inxupt.alkaidchen.com:443/user/${userId}/follows?pageNum=${pageNum}`
-    );
+    const { data } = await axios.get<FollowsInfoFacade>(`user/${userId}/follows`, {
```
返回值的类型应该是 `PagedResponseData<FollowsInfoFacade>`

Marstonchan commented on src/service/post_service.ts:
```diff
@@ -0,0 +1,33 @@
+
+  unFollowUser(userId: number) {
+    const url = `user/follow/${userId}`;
+    axios.delete<void>(url);
```
return

Marstonchan commented on src/service/post_service.ts:
```diff
@@ -0,0 +1,33 @@
+    axios.delete<void>(url);
+  }
+
+  async likeArticle(postId: number) {
```
直接 return axios.xxx 的就不需要 async 了 axios.xxx 本身就是一个 promise

Marstonchan commented on src/types/person_types.ts:
```diff
@@ -0,0 +1,128 @@
+  total: number;
+}
+
+export interface PostInfoFacade {
```
同上

Marstonchan commented on src/types/person_types.ts:
```diff
@@ -0,0 +1,128 @@
+  total: number;
+}
+
+export interface FavoriteInfoFacade {
```
同上

Marstonchan commented on src/types/person_types.ts:
```diff
@@ -0,0 +1,128 @@
+  total: number;
+}
+
+export interface PostsInfoFacade {
```
同上

Marstonchan commented on src/pages/profile/OthersView.vue:
```diff
@@ -0,0 +1,540 @@
+
+const getData = () => {
+  fetchPosts();
+  fetchFans();
```
建议不要同时请求，切换对应的 tab 再进行请求

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
Merge branch 'feature/921l/Othersview' of https://github.com/allspark-studio/inxupt-frontend into feature/921l/Othersview

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
feat(pages): 新增个人页面及他人页面

921l committed:
feat(pages): 新增个人页面及他人页面

Marstonchan committed:
feat(*): 修改问题若干

1. 改为使用绝对路径引入
2. 删除不必要的 props 初始化操作和其他逻辑
3. 新增 UserInfoLoader 用于等待用户信息返回
4. 完善类型定义
5. 规范化文件命名
6. 修复关注接口请求方法错误

921l committed:
fix(pages): 修复关注接口请求错误

Marstonchan committed:
Merge branch 'dev' into feature/921l/Othersview
